src/uwtools/tests/config/test_tools.py

```diff
@@ -670,51 +599,200 @@ def test__print_config_section_yaml_not_dict():
     assert "must be a dictionary" in str(e.value)
 
 
-@pytest.mark.parametrize(
-    "supplemental_configs", [YAMLConfig(config={1: {2: {3: 99}}}), {1: {2: {3: 99}}}]
-)
-def test__realize_config_update(realize_config_testobj, supplemental_configs):
-    assert realize_config_testobj[1][2][3] == 88
-    o = tools._realize_config_update(
-        config_obj=realize_config_testobj,
-        config_fmt="yaml",
-        supplemental_configs=[supplemental_configs],
-    )
-    assert o[1][2][3] == 99
+def test__realize_config_input_setup_ini_cfgobj():
+    data = {"section": {"foo": "bar"}}
+    cfgobj = INIConfig(config=data)
+    input_obj, input_format = tools._realize_config_input_setup(input_config=cfgobj)
+    assert input_obj.data == data
+    assert input_format == FORMAT.ini
 
 
-def test__realize_config_update_noop(realize_config_testobj):
-    assert realize_config_testobj == tools._realize_config_update(
-        config_obj=realize_config_testobj, config_fmt="yaml", supplemental_configs=None
-    )
+def test__realize_config_input_setup_ini_file(tmp_path):
+    data = """
+    [section]
+    foo = bar
+    """
+    path = tmp_path / "config.ini"
+    with open(path, "w", encoding="utf-8") as f:
+        print(dedent(data).strip(), file=f)
+    input_obj, input_format = tools._realize_config_input_setup(input_config=path)
+    assert input_obj.data == {"section": {"foo": "bar"}}
+    assert input_format == FORMAT.ini
 
 
-def test__realize_config_update_file(realize_config_testobj, tmp_path):
-    values = {1: {2: {3: 99}}}
+def test__realize_config_input_setup_ini_stdin(caplog):
+    data = """
+    [section]
+    foo = bar
```

Comment:
One interesting thing I noticed here was that the integer-supporting formats have integers in the tests. I'm curious if it's worth showing here that the integer is processed as a string from ini format...if that's true, of course.

Reply:
Good idea. I just extended this test to include an `int` value.